controllers/common.go

```diff
@@ -25,3 +62,68 @@ func ReconcileBatch(l logr.Logger, reconcileFuncs ...ReconcileFunc) (bool, error
 	}
 	return true, nil
 }
+
+func PopulateResticSecret(vsb *datamoverv1alpha1.VolumeSnapshotBackup, vsr *datamoverv1alpha1.VolumeSnapshotRestore) (*corev1.Secret, error) {
```

Comment:
So we just need label, name and namespace, right ? We can extract that before hand and pass them as parameters, like change the function signature to `PopulateResticSecret(string name, namespace, label) (*corev1.Secret, error)` . This is makes the function CR agnostic and you do not have to pass `nil` as an arg and add a bunch of `If-else` . WDYT ?

Reply:
That works actually. My brain dint think of that even when I changed buildresticsecret func..duh.. new commit coming soon